client/src/components/classesPage/ClassEditView.js

```diff
@@ -1,19 +1,166 @@
-import React from "react";
-import Grid from "@material-ui/core/Grid";
-import { withStyles } from "@material-ui/core/styles";
+import React, { useState, useEffect } from 'react';
+import {
+  Grid,
+  Checkbox,
+  Card,
+  Select,
+  MenuItem,
+  Button
+} from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
 
 const styles = theme => ({
   wrapper: {
+    color: 'white'
+  },
+  cardList: {
+    display: 'flex'
+  },
+  card: {
+    width: 200,
+    height: 200,
+    border: '1px solid white',
+    margin: theme.spacing.unit * 3
   }
 });
 
 function ClassEditView(props) {
+  const { classes } = props;
+  const classId = props.match.params.id;
+  const ax = axios.create({
+    baseURL: 'http://localhost:9000' // development
+  });
+  const [students, setStudents] = useState([]);
+  const [refreshrs, setRefreshrs] = useState([]);
+  const [teacherRefs, setTeacherRefs] = useState([]);
+  const [selectedStudents, setSelectedStudents] = useState([]);
+  // const [classDetails, setClassDetails] = useState([]);
+
+  // get class details on mount
+  useEffect(() => {
+    fetchStudents();
+    fetchRefreshrs();
+    fetchTeacherRefreshrs();
+  }, []);
+
+  useEffect(() => {
+    console.log('students:', students);
+  }, [students]);
```

Comment:
How do these useEffect hooks work? Like instead of having one big one, there's a bunch of them. Do they only run when there's a change to the value you pass at the end, like the `[students]`?

Reply:
Yep, exactly! Passing the array will only run the useEffect when a value in the array changes. Passing the empty array makes it only run when the component first mounts and then again when it unmounts.

Reply:
Thank you.